pandas-stubs/core/indexes/interval.pyi

```diff
@@ -216,46 +250,155 @@ class IntervalIndex(IntervalMixin, ExtensionIndex, Generic[_IntervalT]):
     def get_value(self, series: ABCSeries, key): ...
     @property
     def is_all_dates(self) -> bool: ...
+    # override is due to additional types for comparison
+    # misc is due to overlap with object below
     @overload  # type: ignore[override]
-    def __gt__(self, other: Interval | IntervalIndex) -> np_ndarray_bool: ...  # type: ignore[misc]
+    def __gt__(  # type: ignore[misc]
+        self, other: IntervalT | IntervalIndex[IntervalT]
+    ) -> np_ndarray_bool: ...
+    @overload
+    def __gt__(self, other: pd.Series[IntervalT]) -> pd.Series[bool]: ...  # type: ignore[misc]
     @overload
-    def __gt__(self, other: object) -> bool: ...
+    def __gt__(self, other: object) -> Literal[False]: ...
```

Comment:
I think this (and other comparisons to `object` for `gt`, `ge`, `lt`, `le` should be `Never`.  You can't compare an `Interval` to an arbitrary object.

Reply:
Changed these.